tests/bluetooth/audio/ascs/src/test_ase_state_transition.c

```diff
@@ -110,12 +112,11 @@ static void test_ase_state_transition_after(void *f)
 	err = bt_bap_unicast_server_unregister_cb(&mock_bap_unicast_server_cb);
 	zassert_equal(err, 0, "unexpected err response %d", err);
 
+	/* Sleep to trigger any pending state changes from unregister_cb */
+	k_sleep(K_SECONDS(1));
```

Comment:
If we have to do this in a test, then wouldn't this also imply that the developer would have to do this?

Reply:
Yes and no - `bt_bap_unicast_server_unregister_cb` (today) is releasing the streams, but the finalizing of this is done in the system workqueue. 

You added the following to deal with that in these tests:

```c
	err = bt_bap_unicast_server_unregister();
	while (err != 0) {
		zassert_equal(err, -EBUSY, "unexpected err response %d", err);
		k_sleep(K_MSEC(10));
		err = bt_bap_unicast_server_unregister();
	}
```

But the issue is that for unit tests we don't have a system workqueue thread. 

The `k_sleep` function here is mocked to trigger all system workqueue items. Arguably `k_work_flush` should have been implemented as a mock function to do this, but that felt at bit outside the scope of this PR. 

So yes, an application may need to wait before calling `bt_bap_unicast_server_unregister`, which isn't ideal, but I think there are some upcoming changes to `bt_bap_unicast_server_unregister_cb`, right?

Reply:
Aye, intresting.

Yes, you are right there are some upcoming changes to 'bt_bap_unicast_server_unregister_cb'. Lets make sure all of this is solved in the best possible way then.